### ISSUE TITLE
Dailies don't recognize day of the week?

### ISSUE THREAD

**Issue by gsalvadi** [HabitRPG/habitica#5476]
When I set 'Wed' for a new daily. It was grey. As I tried it, I can understand it shows up today (Wed) when it's set Tuesday. Is that what everyone experience?

**Comment by Alys** [comment]
Do you have a Custom Day Start and if so, what time is it set to, and what was the local time on your PC when this problem happened? 

Does the Daily have a Start Date and what is it? What is the local date where you are?

**Comment by gsalvadi** [comment]
Just happened. I don't have a start date set. But the start date shows today (may be by default). I set Wed and Sat to repeat and switch off other days. It looks grey. My system Date : 24 Jun 15 1:34 PM
When I chose Tuesday, Daily  toggles between on and off (Color and Grey) when I check and uncheck Tuesday. Not Wednesday

**Comment by Alys** [comment]
Sorry, I wasn't clear. The Custom Day Start and the Daily's Start Date are different things. 

You can see what Custom Day Start is set for your account at Settings -> Site. Can you please go there now and check what it's set to, even if you don't recall changing it?

**Comment by gsalvadi** [comment]
You are right. I just changed that to 23 hrs from 0 few hours back. Is that causes this behavior?

**Comment by gsalvadi** [comment]
OK I read 'Complete all your Dailies before changing the Custom Day Start'. Sorry for raising this as an issue. I will take care from this point of time.

**Comment by gsalvadi** [comment]
OK. That's the problem. Sorry for bothering you.

**Comment by Alys** [comment]
That's okay. :) Don't be sorry!

It is the 23 Custom Day Start (CDS) that's causing what you see (although completing your Dailies first would not have made any difference).   A CDS of 23 tells HabitRPG that until 11pm, "today" is actually "yesterday". I've written up something about this elsewhere that explains it better and I'll post that here when I've found it. And probably on the wiki as well where it will be properly useful. :-/

**Comment by gsalvadi** [comment]
OK. Can I have 23 as CDS now? Will it be OK from next day?

**Comment by Alys** [comment]
**UPDATE: I've have added a section to the wiki's [Custom Day Start](http://habitica.wikia.com/wiki/Custom_Day_Start) page to explain the material below. It would be better to read that than this comment.**

It depends on what you want to achieve. It probably will not be what you want, unless your schedule is very different than the average person's.

Here's what I've written [elsewhere](https://github.com/HabitRPG/habitrpg/issues/5357), but I'm pasting it here and adjusting the numbers for your case:

Your custom day start (CDS) is 23:00, and the logic for that means that "Tuesday" actually ends at 11pm on Wednesday. Here are some less brain-twisting examples to help illustrate this:

If your CDS is 0 (the default, midnight), then "Tuesday" ends at midnight between Tuesday and Wednesday.

If your CDS is 1 (1am), then "Tuesday" ends at 1am on Wednesday. That's what people would use if they typically go to sleep after midnight each day.

Following through with the same logic:

If your CDS is 23 (11pm), then "Tuesday" ends at 11pm on Wednesday. Your CDS of 23 tells HabitRPG that before 11pm, you want HabitRPG to consider that you are still on the previous day.

To put it another way, the Custom Day Start tells HabitRPG when each day is meant to start. If the Custom Day Start is 23 (11pm), then Wednesday in HabitRPG does not actually start until 11pm on Wednesday.

What is it you're trying to achieve? What time of day do you want your Dailies to reset? Alternatively, it might help if you tell us what time you normally go to bed and what time you wake up.

**Comment by gsalvadi** [comment]
Now I understand. Changing the CDS unlikely to be helpful I guess. See if you can suggest something. I go to bed at 3:00 AM. After cron, I check what damage I have done to Boss. Accordingly I plan next day. I personally prefer a brutal smash only to jobs that requires great effort physically or mentally. In this way I can see a quest as an opportunity to get rid of procrastinating tasks. In this context,  I start working on next day's schedule after cron, when it's 12 AM. 

I actually want to hit the bed at 10 PM and in a transition. 6 AM is where I came from to 3 AM bed time.

So I thought having the cron 11 PM can help me to check boss damage and to plan next day bit earlier. This way I could move the cron towards 10 PM is what I thought.  :)

**Comment by gsalvadi** [comment]
To put it simple, I want to check the Boss' health bit earlier. Like 10 or 11 PM

**Comment by Alys** [comment]
Damage is always done to the boss at the same time that your Dailies reset, and it's not possible to have them at different times. To make good use of your Dailies, you really do need them to reset when you first use HabitRPG each day, so it is recommended to have CDS set to some time when you are likely to be asleep.

Is your main requirement that you want to see how much boss damage you have already created before it is actually applied to the boss? If so, go to Data -> Data Display Tool, authenticate as it describes on that page, and then look at "Damage to boss" in the dashboard (the series of coloured tiles across the top of the page. Click on that tile to get more information. Is that what you're after?

**Comment by Alys** [comment]
Note that in future, we'll probaby adjust the Custom Day Start feature so that you can specify that "23" means that your day ENDS at 11pm today, rather than STARTING at 11pm today, but I'm not sure when that will happen.

**Comment by gsalvadi** [comment]
@Alys  Thanks. Data Tool suggestion helps to pick hard to dos for next day. And, the CDS doesn't work the way I thought doesn't have any impact on effectiveness of the way the system helps me. Obvious :)

Thanks for your help.

**Comment by Alys** [comment]
No worries. :)
